feature/experimental/p4rt/otg_tests/traceroute_packetin_test/README.md

```diff
@@ -1,34 +1,243 @@
 # P4RT-5.1: Traceroute: PacketIn
 
-
 ## Summary
 
-Verify that Traceroute packets are punted with correct metadata.
+Test FRR behaviors with encapsulation scenarios.
+
+## Topology
+
+ATE port-1 <------> port-1 DUT
+DUT port-2 <------> port-2 ATE
+DUT port-3 <------> port-3 ATE
+DUT port-4 <------> port-4 ATE
+DUT port-5 <------> port-5 ATE
+DUT port-6 <------> port-6 ATE
+DUT port-7 <------> port-7 ATE
+DUT port-8 <------> port-8 ATE
+
+## Baseline setup
+
+*   Setup equivalent to [TE-17.1 vrf_policy_driven_te](https://github.com/openconfig/featureprofiles/blob/main/feature/experimental/gribi/otg_tests/vrf_policy_driven_te/README.md), including GRibi programming.
+
+*   Install a BGP route resolved by ISIS in default VRF to route traffic out of DUT port-8 for 203.0.113.0.
+
+*   Enable the P4RT server on the device.
+
+*   Connect a P4RT client and configure the forwarding pipeline. Install P4RT table entries required for traceroute.
+	These are located in [p4rt_utils.go] (https://github.com/openconfig/featureprofiles/blob/main/internal/p4rtutils/p4rtutils.go)
+	p4rtutils.ACLWbbIngressTableEntryGet(packetIO.GetTableEntry(delete, isIPv4))
 
 
 ## Procedure
 
-*	Connect ATE port-1 to DUT port-1, ATE port-2 to DUT port-2.
-*	TODO: Install the set of routes on the device.
-*	Enable the P4RT server on the device.
-*	Connect a P4RT client and configure the forwarding pipeline. InstallP4RT table 	entries required for traceroute.
-*	Send IPv4 packets from the ATE with TTL=1 and verify that packets with TTL=1 are received by the client.
-*	Send IPv6 packets from the ATE with HopLimit=1 and verify that packets with HopLimit=1 are received by the client.
-*	Verify that the packets have both ingress_singleton_port and egress_singleton_port metadata set.
+At the start of each of the following scenarios, ensure:
+
+*   All ports are up and baseline is reset as above.
+
+Unless otherwise specified, all the tests below should use traffic with
+`dscp_encap_a_1` referenced int he VRF selection policy.
+
+### TODO: Test-1
+
+Tests that traceroute with TTL=1 for a packet that would match the VRF
+selection policy for encap has target_egress_port set based on the
+encap VRF.
+
+*   Send packets to DUT port-1 with outer packet header TTL=1. The outer v4 header has the destination
+    addresses 138.0.11.8. verify that packets with TTL=1 are received by the client.
+
+*   Verify that the punted packets have both ingress_port and target_egress_port metadata set.
+The distribution of packets should have target_egress_port set with port 2 10% of the time, port 3 30%, port 4 60%.
```

Comment:
The Distribution should be port2 1.56%, port3 4.68%, port4 18.75% and port6 75%

Reply:
How do you get that distribution?

Reply:
Based on traffic  & matched policy it will hit vrf EncapVRFA which will do encap with IP 203.0.113.1 & 203.0.113.2 which will do load share to port2, 3, 4 & 6.

Policy:
pfRule13 := &policyFwRule{SeqId: 13, family: "ipv4", dscpSet: []uint8{dscpEncapA1, dscpEncapA2},
		ni: niEncapTeVrfA}

Reply:
Done,.